variables.tf

```
@@ -4,30 +4,135 @@
   default     = "us-east-1"
+  description = "The AWS region to deploy into (e.g. us-east-1)."
+  type        = string
+}
+
+variable "ec2_read_role_name" {
+  default     = "EC2ReadOnly"
+  description = "The name of the IAM role that allows read access to the necessary EC2 attributes.  Note that this role must exist in each account that you want to query."
+  type        = string
+}
+
+variable "extraorg_account_ids" {
+  default     = []
+  description = "A list of AWS account IDs corresponding to \"extra\" accounts that you want to query for egress IPs to publish."
+  type        = list(string)
+}
+
+variable "file_configs" {
+  default     = []
+  type        = list(object({ app_regex = string, description = string, filename = string, static_ips = list(string) }))
+  description = "A list of objects that define the files to be published.  \"app_regex\" specifies a regular expression matching the application name (based on the var.application_tag).  \"description\" is the description of the published file.  \"filename\" is the name to assign the published file.  \"static_ips\" is a list of CIDR blocks that will always be included in the published file.  An example file configuration looks like this: `[{\"app_regex\": \".*\", \"description\": \"This file contains a list of all public IP addresses to be published.\", \"filename\": \"all.txt\",  \"static_ips\": []}, {\"app_regex\": \"^Vulnerability Scanning$\", \"description\": \"This file contains a list of all IPs used for Vulnerability Scanning.\", \"filename\": \"vs.txt\",  \"static_ips\": [\"192.168.1.1/32\", \"192.168.2.2/32\"]}]`"
+}
+
+variable "file_header" {
+  default     = "###\n# https://{domain}/{filename}\n# {timestamp}\n# {description}\n###\n"
+  description = "The header template for each published file.  The following variables are available within the template: {domain} - the domain where the published files are located, {filename} - the name of the published file, {timestamp} - the timestamp when the file was published, {description} - the description of the published file"
+  type        = string
+}
+
+variable "lambda_function_description" {
+  default     = "Lambda function to publish egress IP addresses to an S3 bucket configured with a CloudFront distribution for HTTPS access."
+  description = "The description of the Lambda function."
+  type        = string
+}
+
+variable "lambda_function_name" {
+  default     = "publish-egress-ip"
+  description = "The name of the Lambda function to publish egress IP addresses."
+  type        = string
+}
+
+variable "lambda_schedule_interval" {
+  default     = 60
+  description = "The number of minutes between scheduled runs of the Lambda function to publish egress IP addresses.  This value must be an integer greater than 0."
+  type        = number
+
+  validation {
+    condition     = try(parseint(tostring(var.lambda_schedule_interval), 10), null) == var.lambda_schedule_interval && var.lambda_schedule_interval > 0
```

Since the value has already been converted to a numeric value I think this makes more sense than a series of type conversions. Alternatively I think `can(parseint(tostring(var.lambda_schedule_interval), 10))` is a better implementation for the first test. I subbed in `alltrue()` because I think it makes for a cleaner representation of a series of conditions.

```suggestion
    condition     = alltrue([floor(var.lambda_schedule_interval) == var.lambda_schedule_interval, var.lambda_schedule_interval > 0])
```

---

I think your suggestion is absolutely an improvement over what I had initially, thanks!

BTW, I didn't know that `can()` function existed- that is what I was looking for earlier when I used `try()`.  Good to know!